content/en/agent/guide/heroku-ruby.md

```diff
@@ -225,32 +217,305 @@ Collector
+
+```ruby
+    require 'datadog/statsd' # gem 'dogstatsd-ruby'
+
+    Sidekiq::Pro.dogstatsd = ->{ Datadog::Statsd.new('localhost', 8125, namespace:'sidekiq') }
+
+    Sidekiq.configure_server do |config|
+      config.server_middleware do |chain|
+        require 'sidekiq/middleware/server/statsd'
+        chain.add Sidekiq::Middleware::Server::Statsd
+      end
+    end
+```
+
+If you are using Sidekiq Enterprise and would like to collect historical metrics, include this line as well:
+
+```ruby
+      Sidekiq.configure_server do |config|
+        # history is captured every 30 seconds by default
+        config.retain_history(30)
+      end
+```
+
+Add the following to your [`datadog/prerun.sh`][14] script:
+
+```
+cat << 'EOF' >> "$DATADOG_CONF"
+
+dogstatsd_mapper_profiles:
+  - name: sidekiq
+    prefix: "sidekiq."
+    mappings:
+      - match: 'sidekiq\.sidekiq\.(.*)'
+        match_type: "regex"
+        name: "sidekiq.$1"
+      - match: 'sidekiq\.jobs\.(.*)\.perform'
+        name: "sidekiq.jobs.perform"
+        match_type: "regex"
+        tags:
+          worker: "$1"
+      - match: 'sidekiq\.jobs\.(.*)\.(count|success|failure)'
+        name: "sidekiq.jobs.worker.$2"
+        match_type: "regex"
+        tags:
+          worker: "$1"
+EOF
+```
+
+Deploy to Heroku:
+
+```shell
+# Deploy to Heroku
+git add .
+git commit -m "Enable sidekiq integration"
+git push heroku main
+```
+
+Once the build finishes, the Datadog Agent starts the Sidekiq check. Run the Datadog Agent status as explained in the [appendix section](#appendix-getting-the-datadog-agent-status) to make sure the Sidekiq check is running correctly.
+
+### Memcached
+
+Memcached is a distributed memory object caching system that is very popular in Rails applications. For this example, we will attach the [Heroku Memcached Cloud Addon][17] to our Heroku application:
+
+```
+heroku addons:create memcachedcloud:30
+```
+
+To check that Memcached has been succesfully attached to your application, you can run the following command:
+
+```shell
+heroku addons | grep -A2 memcachedcloud
+```
+
+You should get an output similar to this one:
+
+```shell
+memcachedcloud (memcachedcloud-fluffy-34783)   30         free   created
+ └─ as MEMCACHEDCLOUD
+```
+
+And retrieve the credentials from HEROKU by running the following command:
+
+```shell
+heroku config | grep MEMCACHEDCLOUD
+```
+
+Create a configuration file called `/datadog/conf.d/mcache.yaml` at the root of your application replacing your host, port, username and password with the information you got in the previous command:
+
+```yaml
+instances:
+  - url: <YOUR_MCACHE_HOST> 
+    port: <YOUR_MCACHE_PORT>
+    username: <YOUR_MCACHE_USERNAME>
+    password: <YOUR_MCACHE_PASSWORD>
+```
+
+Instead of manually replacing and harcoding the configuration, you can also set up your Memcached integration based on Heroku environment variables, using the [prerun script][14] to replace those values before starting the Datadog agent:
```

Comment:
```suggestion
Instead of manually updating the configuration, you can set up your Memcached integration based on Heroku environment variables using the [prerun script][14] to replace those values before starting the Datadog Agent:
```